From ce61970a61d0fb51b679ea1a4abc6d13a96b9d02 Mon Sep 17 00:00:00 2001
From: Daniel Watkins <daniel.watkins@canonical.com>
Date: Wed, 18 Feb 2015 18:09:34 +0000
Subject: [PATCH 3/8] Failing to fetch a CloudStack password should never fail
 the whole DS.

There might be some CloudStack deployments without the :8080 password
server, and there's no reason the rest of the data source can't be used
for them.
---
 cloudinit/sources/DataSourceCloudStack.py | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)

diff --git a/cloudinit/sources/DataSourceCloudStack.py b/cloudinit/sources/DataSourceCloudStack.py
index a4c7dbdc4f6f..211893573235 100644
--- a/cloudinit/sources/DataSourceCloudStack.py
+++ b/cloudinit/sources/DataSourceCloudStack.py
@@ -115,15 +115,21 @@ class DataSourceCloudStack(sources.DataSource):
                                                       self.metadata_address)
             LOG.debug("Crawl of metadata service took %s seconds",
                       int(time.time() - start_time))
-            set_password = self.get_password()
-            if set_password:
-                self.cfg = {
-                    'ssh_pwauth': True,
-                    'password': set_password,
-                    'chpasswd': {
-                        'expire': False,
-                    },
-                }
+            try:
+                set_password = self.get_password()
+            except Exception:
+                util.logexc(LOG,
+                            'Failed to fetch password from virtual router %s',
+                            self.vr_addr)
+            else:
+                if set_password:
+                    self.cfg = {
+                        'ssh_pwauth': True,
+                        'password': set_password,
+                        'chpasswd': {
+                            'expire': False,
+                        },
+                    }
             return True
         except Exception:
             util.logexc(LOG, 'Failed fetching from metadata service %s',
-- 
2.11.0

